api/src/wfl/module/covid.clj

```diff
@@ -263,6 +236,35 @@
                     (->> (jdbc/insert! tx terra-executor-table)))]
     [terra-executor-type (-> items first :id)]))
 
+(defn import-snapshot!
+  "Return snapshot reference for SNAPSHOT_ID imported to WORKSPACE."
+  [{:keys [workspace] :as _executor}
+   {:keys [snapshot_id] :as _snapshot}]
+  (rawls/create-snapshot-reference workspace snapshot_id))
+
+(defn ^:private update-terra-executor
+  "IN PROGRESS:
+  Create new submission from new SOURCE snapshot.
+  Add its workflows to the workflow queue.
+  Return updated EXECUTOR."
+  [source executor]
+  (letfn [(create-submission! [executor reference]
+            ;; Tom to implement?
+            ;; Create submission from snapshot reference.
+            )
+          (push-to-executor! [source executor submission]
+            (pop-queue! source)
+            ;; Write new records to executor queue (:details executor)
+            ;; for each workflow in submission.
+            )
+          (update-executor! [executor]
+            (assoc executor :updated (OffsetDateTime/now)))]
```

Comment:
Not sure if we want to do this anymore, we may be happy with setting `:updated` on the queue objects.